app/controllers/application_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class ApplicationController < ActionController::Base
+  helper_method :current_user, :logged_in?
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def require_user
+    return if logged_in?
+
+    flash[:danger] = 'You must log in to perform that action'
+    redirect_to root_path
+  end
+
+  def create_token_and_save(user)
+    user.create_token_and_save
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:gen_token] = user.gen_token
+  end
```

Comment:
[REQUIRED]
* We need to set the cookies every time the user logs in.
We can just update the `remeber_digest` attribute of the user and set a new token ccokie each time a user logs in.
The method should look similar ti this 
```ruby
def sign_in
  gen_token = User.new_token
  user.update_attribute(:remember_digest, User.encrypted(gen_token)
  cookies.permanent[:gen_token] = gen_token
  current_user = user
end
```